catalog/plugin/catalog-plugin-duplication/src/test/java/ddf/catalog/metacard/duplication/TestDuplicationValidator.java

```diff
@@ -0,0 +1,258 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.metacard.duplication;
+
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsCollectionContaining.hasItems;
+import static org.hamcrest.core.IsNot.not;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.opengis.filter.Filter;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.validation.ValidationException;
+import ddf.catalog.validation.report.MetacardValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestDuplicationValidator {
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private FilterBuilder mockFilterBuilder;
+
+    @Mock
+    private CatalogFramework mockFramework;
+
+    private DuplicationValidator validator;
+
+    private MetacardImpl matchingMetacard;
+
+    private MetacardImpl testMetacard;
+
+    private static final String ID = "matching metacard id";
+
+    private ArgumentCaptor<String> attributeValueCaptor = ArgumentCaptor.forClass(String.class);
+
+    private static final String TAG1 = "1";
+
+    private static final String TAG2 = "2";
+
+    private Set tags = new HashSet<>(Arrays.asList(TAG1, TAG2));
+
+    @Before
+    public void setup()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+
+        when(mockFilterBuilder.attribute(anyString())
```

Comment:
Would it be possible to do something like:
```verify(mockFilterBuilder, times(number of times))
    .attribute(anyString()).equalTo().text(attributeValueCaptor.capture());```
in the test in which you're using the captured values? Using an `ArgumentCaptor` inside a stub makes the test difficult to understand.